### PR TITLE
feat: Created text input

### PR DESCRIPTION
**Pull request by mikecoomber**
refactor: Refactored date and time inputs to use text input

### CODE REVIEW

File: lib/src/components/time_input/time_input.dart [hunk 1]
```diff
@@ -220,9 +169,21 @@ class ZetaTimeInputState extends State<ZetaTimeInput> {
       context: context,
       initialTime: _value ?? TimeOfDay.now(),
       builder: (BuildContext context, Widget? child) {
-        return MediaQuery(
-          data: MediaQuery.of(context).copyWith(alwaysUse24HourFormat: !_use12Hr),
-          child: child!,
+        return Theme(
+          data: Theme.of(context).copyWith(
+            timePickerTheme: TimePickerThemeData(
```
Comment (thelukewalton) [inline lib/src/components/time_input/time_input.dart:174]: something not right here with text color when focused
image

File: example/lib/pages/components/date_input_example.dart [hunk 2]
```diff
@@ -52,10 +43,10 @@ class _DateInputExampleState extends State<DateInputExample> {
               padding: const EdgeInsets.all(20),
               child: ZetaDateInput(
                 label: 'Label',
-                hint: 'Default hint text',
-                errorText: 'Oops! Error hint text',
+                hintText: 'Default hintText text',
```
Comment (thelukewalton) [inline example/lib/pages/components/date_input_example.dart:46]: why the double text?

File: example/widgetbook/pages/components/date_input_widgetbook.dart [hunk 3]
```diff
@@ -32,13 +32,12 @@ Widget dateInputUseCase(BuildContext context) {
           child: ZetaDateInput(
             size: size,
             rounded: rounded,
-            enabled: enabled,
+            disabled: enabled,
```
Comment (thelukewalton) [inline example/widgetbook/pages/components/date_input_widgetbook.dart:35]: this is wrong way around

File: lib/src/components/date_input/date_input.dart [hunk 4]
```diff
@@ -1,319 +1,314 @@
...
+
+    final result = await showDatePicker(
+      context: context,
+      firstDate: firstDate,
+      lastDate: lastDate,
+      initialDate: initialDate,
+      builder: (BuildContext context, Widget? child) {
+        return Theme(
+          data: Theme.of(context).copyWith(
+            datePickerTheme: const DatePickerThemeData(
```
Comment (thelukewalton) [inline lib/src/components/date_input/date_input.dart:183]: personal preference: i think this divider is too dark.
also it is really rounded, should / can this have a sharp variant?
image

File: lib/src/components/date_input/date_input.dart [hunk 5]
```diff
@@ -1,319 +1,314 @@
...
   }
 }
 
-class _ZetaDateInputState extends State<ZetaDateInput> {
-  final _controller = TextEditingController();
-  late ZetaWidgetSize _size;
-  late final String _hintText;
+/// State for [ZetaDateInput]
+class ZetaDateInputState extends State<ZetaDateInput> implements ZetaFormFieldState {
+  // TODO(mikecoomber): add AM/PM selector inline.
```
Comment (thelukewalton) [inline lib/src/components/date_input/date_input.dart:92]: todo

File: lib/src/components/text_input/text_input.dart [hunk 6]
```diff
@@ -0,0 +1,446 @@
...
+  }
+
+  OutlineInputBorder get _baseBorder => OutlineInputBorder(
+        borderRadius: widget.rounded ? ZetaRadius.minimal : ZetaRadius.none,
+        borderSide: BorderSide(color: _hovered ? _colors.borderSelected : _colors.borderSubtle),
+      );
+
+  OutlineInputBorder get _focusedBorder => _baseBorder.copyWith(
+        borderSide: BorderSide(color: _colors.primary.shade50, width: ZetaSpacing.x0_5),
+      ); // TODO(mikecoomber): change to colors.borderPrimary when added
```
Comment (thelukewalton) [inline lib/src/components/text_input/text_input.dart:214]: todo

File: lib/src/components/text_input/text_input.dart [hunk 7]
```diff
@@ -0,0 +1,446 @@
...
+              fillColor: _backgroundColor,
+              enabledBorder: _baseBorder,
+              disabledBorder: _baseBorder,
+              focusedBorder: _focusedBorder,
+              focusedErrorBorder: _errorBorder,
+              errorBorder: widget.disabled ? _baseBorder : _errorBorder,
+              hintText: widget.placeholder,
+              errorText: _errorText,
+              hintStyle: _baseTextStyle,
+              errorStyle: const TextStyle(height: 0.001, color: Colors.transparent),
```
Comment (thelukewalton) [inline lib/src/components/text_input/text_input.dart:309]: interesting...
could you use 0 to fully hide it?

File: lib/src/components/text_input/text_input.dart [hunk 8]
```diff
@@ -0,0 +1,446 @@
...
+    if (requirementLevel == ZetaFormFieldRequirement.optional) {
+      requirementWidget = Text(
+        '(optional)', // TODO(UX-1003): needs localizing.
+        style: textStyle.copyWith(color: disabled ? colors.textDisabled : colors.textSubtle),
+      );
+    } else if (requirementLevel == ZetaFormFieldRequirement.mandatory) {
+      requirementWidget = Text(
+        '*',
+        style: ZetaTextStyles.labelIndicator.copyWith(
+          color: disabled ? colors.textDisabled : colors.error, // TODO(mikecoomber): change to textNegative when added
```
Comment (thelukewalton) [inline lib/src/components/text_input/text_input.dart:351]: todo

**Review by thelukewalton — CHANGES_REQUESTED** [review]

File: lib/src/components/date_input/date_input.dart (hunk 5 above)
Comment (mikecoomber) [reply lib/src/components/date_input/date_input.dart:92]: Was waiting on designs, should I just freestyle it?

File: lib/src/components/text_input/text_input.dart (hunk 6 above)
Comment (mikecoomber) [reply lib/src/components/text_input/text_input.dart:214]: this color isn't in Zeta yet, hence the todo

File: lib/src/components/text_input/text_input.dart (hunk 8 above)
Comment (mikecoomber) [reply lib/src/components/text_input/text_input.dart:351]: this color isn't in Zeta yet, hence the todo

File: lib/src/components/text_input/text_input.dart (hunk 7 above)
Comment (mikecoomber) [reply lib/src/components/text_input/text_input.dart:309]: Weirdly that wasn't actually hiding the text in some cases, found this as a workaround

File: lib/src/components/date_input/date_input.dart (hunk 4 above)
Comment (mikecoomber) [reply lib/src/components/date_input/date_input.dart:183]: yep will do. Do you want me to reduce the border radius on this when the component is rounded?

File: example/lib/pages/components/date_input_example.dart (hunk 2 above)
Comment (mikecoomber) [reply example/lib/pages/components/date_input_example.dart:46]: a find and replace misshap

File: lib/src/components/date_input/date_input.dart (hunk 4 above)
Comment (thelukewalton) [reply lib/src/components/date_input/date_input.dart:183]: yeah i think so

File: lib/src/components/date_input/date_input.dart (hunk 5 above)
Comment (thelukewalton) [reply lib/src/components/date_input/date_input.dart:92]: I see, sorry wasnt looking close enough. In situations like this I have been creating a Jira ticket and replacing 'mikecoomber' with the ticket number

**Review by thelukewalton — APPROVED** [review]

**Pull request merged**